### PR TITLE
Mitigate XSS vulnerabilities

### PR DESCRIPTION
**Pull request by krkshitij**

### CODE REVIEW

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 1]
```diff
@@ -710,6 +710,7 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
             onMouseMove={this._onRectMouseMove}
             onMouseOut={this._onRectMouseOut}
             onMouseOver={this._onRectMouseMove}
+            // FIXME: XSS vulnerability: https://stackoverflow.com/collectives/articles/75138459/mitigating-xss-attacks-in-react-applications#heading-xss-via-user-defined-props
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:713]: don't add todos in the existing code. Put these in the one note

**Comment by AtishayMsft** [comment]
I did not get the purpose of this PR. Is this only to track the issues?

**Comment by krkshitij** [comment]
> I did not get the purpose of this PR. Is this only to track the issues?

This is still a WIP. I have identified some bad practices (non-critical vulnerabilities) and will be fixing them one by one.

**Comment by AtishayMsft** [comment]
> > I did not get the purpose of this PR. Is this only to track the issues?
> 
> This is still a WIP. I have identified some bad practices (non-critical vulnerabilities) and will be fixing them one by one.

Handle possible issues in the declarative chart transformation logic only.
Other vulnerabilities in the existing product will be handled separately.

**Comment by krkshitij** [comment]
> > > I did not get the purpose of this PR. Is this only to track the issues?
> > 
> > 
> > This is still a WIP. I have identified some bad practices (non-critical vulnerabilities) and will be fixing them one by one.
> 
> Handle possible issues in the declarative chart transformation logic only. Other vulnerabilities in the existing product will be handled separately.

I tried injecting scripts into our charts through Plotly json, but React handled them well and prevented their execution.
Here's an example: 

image

So far, we are not using any of the props marked in this PR, so there is nothing immediate to fix.

However, to mitigate potential risks, I explored two ways:
- Use a third-party library like DOMPurify to sanitize Plotly properties before mapping them to chart props.
- Rely on React's auto-escaping to prevent malicious content from getting injected into the DOM and focus on resolving any additional vulnerabilities in our implementation. (I chose this approach.)

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 2]
```diff
@@ -704,13 +705,15 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
+            strokeWidth={points[index].lineOptions?.strokeWidth ?? 3}
             stroke={this._colors[index]}
             opacity={this._getLineOpacity(points[index]!.legend)}
             onMouseMove={this._onRectMouseMove}
             onMouseOut={this._onRectMouseOut}
             onMouseOver={this._onRectMouseMove}
-            {...points[index]!.lineOptions}
+            strokeDasharray={points[index].lineOptions?.strokeDasharray}
+            strokeDashoffset={points[index].lineOptions?.strokeDashoffset}
+            strokeLinecap={points[index].lineOptions?.strokeLinecap ?? 'round'}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:716]: Is 'round' the current default behavior.

File: packages/charts/react-charting/src/utilities/utilities.ts [hunk 3]
```diff
@@ -1384,3 +1384,22 @@ export const formatDate = (date: Date, useUTC?: boolean) => {
   const timeFormat = useUTC ? d3UtcFormat : d3TimeFormat;
   return timeFormat('%-e %b %Y, %H:%M')(date) + (useUTC ? ' GMT' : '');
 };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const getSafeProps = (props: Record<string, any> | undefined) => {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/utilities/utilities.ts:1389]: Can you see if this code is better and works as expected:

function getSecureProps(props: Record<string, unknown> = {}): Record<string, unknown> {
  const { dangerouslySetInnerHTML, ...result } = props;
  return result;
}

**Comment by AtishayMsft** [comment]
Target master for this fix.

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 2 above)
Comment (krkshitij) [reply packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:716]: It is the default value in LineChart but not in AreaChart. Removed it.

File: packages/charts/react-charting/src/utilities/utilities.ts (hunk 3 above)
Comment (krkshitij) [reply packages/charts/react-charting/src/utilities/utilities.ts:1389]: This looks better. Replaced the implementation.

**Comment by krkshitij** [comment]
> Target master for this fix.

Opened a [new PR](https://github.com/microsoft/fluentui/pull/33376) targeting master. Closing this one.

**Pull request closed without merging**